link-verifier/verify-links.py

```diff
@@ -304,47 +306,53 @@ def main():
     parser.add_argument("-A", "--allowlist-file", action="store", dest="allowlist", help="Path to file containing list of allowed URLs.")
     parser.add_argument("-n", "--num-processes", action="store", type=int, default=4, help="Number of processes to run in parallel")
     parser.add_argument("-k", "--keep", action="store_true", default=False, help="Keep temporary files instead of deleting")
+    parser.add_argument("-u", "--user-agent", action="store", dest="user_agent", default=None, help="User agent to use for HTTP requests")
     parser.add_argument("-v", "--verbose", action="store_true", default=False, help="Print all links tested")
     args = parser.parse_args()
 
     html_file_list = []
     broken_links = []
-    file_list = []
-    link_list = []
+    md_file_list = []
+    link_set = set()
     exclude_dirs = [dir.lower() for dir in args.exclude_dirs] if args.exclude_dirs else []
 
-    # If any explicit files are passed, add them to file_list.
+    if args.user_agent != None:
+        http_headers.update({ 'User-Agent': args.user_agent })
+
+    if args.verbose:
+        print("Using User-Agent: {}".format(http_headers['User-Agent']))
+
+
+    # If any explicit files are passed, add them to md_file_list.
     if args.files is not None:
-        file_list = args.files
+        md_file_list = args.files
     elif args.test_markdown:
         # Obtain list of Markdown files from the repository (along with excluding passed directories).
         for root, dirs, files in os.walk("./"):
             # Prune dirs to remove exclude directories, if passed as command line input, from search.
             dirs[:] = [dir for dir in dirs if dir.lower() not in exclude_dirs]
-            file_list += [os.path.join(root, f) for f in files if re.search(MARKDOWN_SEARCH_TERM, f, re.IGNORECASE)]
-
-    if args.verbose:
-        print(file_list)
+            md_file_list += [os.path.join(root, f) for f in files if re.search(MARKDOWN_SEARCH_TERM, f, re.IGNORECASE)]
```

Comment:
Why was printing of the markdown files list removed when verbosity is turned on?

Reply:
It wasn't labelled and was a bit confusing. I can add it back and label it if you would like.

Reply:
Yes, it will be valuable to print all the files included in the scanning as part of the verbose (or dev mode) runs